### PR TITLE
Refactor: Remove code duplication

### PR DESCRIPTION
**Pull request by kristinpeterson**
The [JsonRequest](https://github.com/attm2x/m2x-android/blob/master/app/src/main/java/m2x/att/client/network/JsonRequest.java) file has some deal of repeated code for handling errors, check [PR #1](1).

Move that code to a helper to avoid said duplication.

### CODE REVIEW

**Comment by kristinpeterson** [comment]
@fsaravia this is ready for review :smiley_cat:

File: app/src/main/java/com/att/m2x/android/network/JsonRequest.java [hunk 1]
```diff
@@ -330,85 +158,104 @@ public static void makeDeleteRequest(final Context context,
...
+        //Save response
+        APISharedPreferences.saveLastResponse(context,apiResponse);
+        listener.onRequestError(apiResponse,requestCode);
+    }
 
+    private static Map<String, String> getHeaders(Context context) throws AuthFailureError {
+        Map<String, String> params = new HashMap<String, String>();
+        params.put("Content-Type", "application/json");
+        params.put("X-M2X-KEY", APISharedPreferences.getApiKey(context));
+        params.put("User-agent", "M2X-Android/2.0.0 java/21 (".
```
Comment (fsaravia) [inline app/src/main/java/com/att/m2x/android/network/JsonRequest.java]: Probably out of scope for this PR, but we should move this one to a constant, what do you think?

**Comment by fsaravia** [comment]
:+1: :shipit:

File: app/src/main/java/com/att/m2x/android/network/JsonRequest.java (hunk 1 above)
Comment (kristinpeterson) [inline app/src/main/java/com/att/m2x/android/network/JsonRequest.java]: I agree. I've added a USER_AGENT constant to the Constants class, and updated the version number in the User Agent string as it was outdated.

I was thinking of saving a version-bump/release for the updates that will be made to the library this month (which will result in a bump of the MINOR version num), unless think differently. Let me know!

Comment (fsaravia) [inline app/src/main/java/com/att/m2x/android/network/JsonRequest.java]: > I was thinking of saving a version-bump/release for the updates that will be made to the library this month (which will result in a bump of the MINOR version num), unless think differently. Let me know!

Nope, 100% agreed!

**Pull request merged**